android/KMEA/app/src/main/java/com/tavultesoft/kmea/KeyboardPickerActivity.java

```diff
@@ -684,31 +566,24 @@ protected static void notifyLexicalModelsUpdate(Context context) {
     storage.lexicalModels.notifyDataSetChanged();
   }
 
-  protected static ArrayList<HashMap<String, String>> getKeyboardsList(Context context) {
-    return getList(context, KMManager.KMFilename_KeyboardsList);
+  protected static List<Keyboard> getKeyboardsList(Context context) {
+    return KeyboardController.getInstance().get();
   }
 
   protected static ArrayList<HashMap<String, String>> getLexicalModelsList(Context context) {
     return getList(context, KMManager.KMFilename_LexicalModelsList);
   }
 
   protected static boolean containsKeyboard(Context context, String keyboardKey) {
-    if (keyboardsList == null) {
-      keyboardsList = getKeyboardsList(context);
-    }
+    List<Keyboard> keyboardsList = KeyboardController.getInstance().get();
 
-    String kbKeys = "";
-    if (keyboardsList != null) {
-      int length = keyboardsList.size();
-      for (int i = 0; i < length; i++) {
-        kbKeys += String.format("%s_%s", keyboardsList.get(i).get(KMManager.KMKey_LanguageID), keyboardsList.get(i).get(KMManager.KMKey_KeyboardID));
-        if (i < length - 1) {
-          kbKeys += ",";
-        }
+    for(Keyboard k : keyboardsList) {
```

Comment:
I'd use a `KeyboardList.getIndexOfKey()` function and if it returns `-1`, return `false`.

Reply:
I've added `KeyboardController.keyboardExists()`